python/device/device.py

```diff
@@ -263,99 +264,113 @@ def update_initial_reported_properties(self):
-        self.client.patch_twin_reported_properties(reported_properties)
-
-        logger.info("sending pairingRequest message")
-        pairing_request_message = Message(json.dumps({"thief": {"cmd": "pairingRequest"}}))
-        pairing_request_message.content_type = "application/json"
-        pairing_request_message.content_encoding = "utf-8"
-        self.client.send_message(pairing_request_message)
-
-        start_time = time.time()
-        while time.time() - start_time < self.config.pairing_request_timeout_interval:
-            logger.info("Waiting for service app to claim this device")
-            patch = self.client.receive_twin_desired_properties_patch(
-                timeout=self.config.pairing_request_send_interval
-            )
+        while not self.done.isSet():
+            worker_thread_info.watchdog_epochtime = time.time()
 
-            if patch and "thief" in patch:
-                logger.info("Got patch")
-                thief = patch["thief"]
-                service_app_run_id = thief.get("serviceAppRunId", None)
-                if (thief.get("pairingId", None) == pairing_id) and service_app_run_id:
-                    logger.info(
-                        "Service app {} claimed this device instance".format(service_app_run_id)
-                    )
-                    self.service_app_run_id = service_app_run_id
-                    reported_properties["thief"]["serviceAppRunId"] = service_app_run_id
-                    self.client.patch_twin_reported_properties(reported_properties)
-                    return
-                else:
-                    logger.info("patch received, but serviceAppRunId not set")
+            try:
+                msg = self.incoming_pairing_message_queue.get(timeout=1)
+            except queue.Empty:
+                msg = None
 
-            logger.info("Still no assigned service instance. Sending new pairingRequest message")
-            self.client.send_message(pairing_request_message)
+            # We trigger a pairing operation by pushing this string into our incoming queue.
+            # Not a great design, but it lets us group this functionality into a single thread
+            # without adding another signalling mechanism.
+            if msg == "START_PAIRING":
+                logger.info("Starting pairing operation")
+                currently_pairing = True
+                pairing_start_epochtime = time.time()
+                pairing_last_request_epochtime = 0
+                self.pairing_id = str(uuid.uuid4())
+                self.service_run_app_id = None
+                # set msg to None to trigger the block that sends the first pairingRequest message
+                msg = None
 
-        raise Exception("timeout waitig for service to set serviceAppRunId")
+            if not msg and currently_pairing:
+                if (
+                    time.time() - pairing_start_epochtime
+                ) > self.config.pairing_request_timeout_interval:
```

Comment:
I'm no python expert, but it looks like this ending parenthesis should be just before the colon

Reply:
It looks like the if statement condition is meant to be comparing the difference in time to that constant. I just don't understand Python.